frontend/src/auth/pages/RegisterPage.tsx

```diff
@@ -1,3 +1,32 @@
+import { z } from 'zod'
+import { useForm, SubmitHandler } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+
+const SignUpSchema = z.object({
+  email: z.string().email(),
+  password: z.string().min(3).max(20),
+})
+type SignUpSchemaType = z.infer<typeof SignUpSchema>
+
 export function RegisterPage() {
-  return <div>Register</div>
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<SignUpSchemaType>({ resolver: zodResolver(SignUpSchema) })
+
+  const onSubmit: SubmitHandler<SignUpSchemaType> = (data) => console.log(data)
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)} className="form">
+      <input className="input" placeholder="email" {...register('email')} />
```

Comment:
replace inputs with shadcn/ui. result be something like that
![image](https://github.com/user-attachments/assets/98223330-fad6-46aa-a897-005b97ee9ab9)